src/python/pants/core/util_rules/archive.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+from pants.engine.fs import Digest, DirectoryWithPrefixToStrip, Snapshot
+from pants.engine.process import Process, ProcessResult
+from pants.engine.rules import RootRule, rule
+from pants.engine.selectors import Get
+
+
+@dataclass(frozen=True)
+class MaybeExtractable:
+    """A newtype to work around rule graph resolution issues.
```

Comment:
Sorry for the trouble. Would the following work?:
```
... = await Get[Digest](ExtractableArchive(digest))
```

A better link might be to https://github.com/pantsbuild/pants/issues/7710.

Reply:
You're right that that is the right link. But no, your suggestion is what didn't work originally, hence this newtype...